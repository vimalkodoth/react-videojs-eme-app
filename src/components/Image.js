import React, { Component } from "react";
import { css } from "@emotion/core";
import spacerImage from "./../static/img/spacer-image.png";
import { string } from "prop-types";
class Image extends Component {
    static propTypes = {
        src: string
    };
    static defaultProps = {
        src: spacerImage
    };
    state = {
        loaded: false,
        defaultSrc: spacerImage
    };

    imageLoaded = () => {
        this.setState((state) => ({
            ...state,
            loaded: true
        }));
    };

    render() {
        const { src } = this.props;
        const { defaultSrc, loaded } = this.state;
        return (
            <React.Fragment>
                <img
                    src={`${src}`}
                    onLoad={this.imageLoaded}
                    css={css`
                        display: ${loaded ? "block" : "none"};
                    `}
                />
                <img
                    src={`${defaultSrc}`}
                    css={css`
                        display: ${!loaded ? "block" : "none"};
                    `}
                />
            </React.Fragment>
        );
    }
}

export default Image;
