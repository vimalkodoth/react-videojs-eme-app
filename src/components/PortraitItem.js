import React from "react";
import Image from "./Image";
import { Link } from "react-router-dom";
import { css } from "@emotion/core";
import PropTypes, { string, object } from "prop-types";

class PortraitItem extends React.Component {
    static propTypes = {
        item: object,
        to: PropTypes.oneOfType([string, object])
    };
    static defaultProps = {
        item: {},
        to: {}
    };
    render() {
        const { item, to } = this.props;

        return (
            <div className="item" css={PortraitItemStyles}>
                <Link to={to}>
                    <Image src={item.images.artwork} />
                </Link>
            </div>
        );
    }
}

export default PortraitItem;

const PortraitItemStyles = css`
    &.item {
        width: 100%;
        border: 5px solid;
        border-color: transparent;
        &:hover {
            border-color: yellow;
        }
        a {
            display: flex;
        }
        img {
            min-width: 100%;
            max-width: 100%;
        }
    }
`;
