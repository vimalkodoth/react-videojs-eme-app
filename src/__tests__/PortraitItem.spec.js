import React from "react";
import { shallow } from "enzyme";
import PortraitItem from "../components/PortraitItem";
import list from "./json/list.json";
import { Link } from "react-router-dom";
import Image from "../components/Image";

describe("PortraitItem", () => {
    it("renders correctly", () => {
        const movieItem = list.data.contents.data[0];
        const component = shallow(
            <PortraitItem
                key={movieItem.id}
                item={movieItem}
                to={{
                    pathname: `/details/${movieItem.id}`,
                    state: { from: location.pathname }
                }}
            ></PortraitItem>
        );
        expect(component).toMatchSnapshot();
    });

    it("renders Image inside Link Item", () => {
        const movieItem = list.data.contents.data[0];
        const component = shallow(
            <PortraitItem
                key={movieItem.id}
                item={movieItem}
                to={{
                    pathname: `/details/${movieItem.id}`,
                    state: { from: location.pathname }
                }}
            ></PortraitItem>
        );
        expect(component.find(Link).length).toEqual(1);
        expect(component.find(Link).find(Image).length).toEqual(1);
    });
});
