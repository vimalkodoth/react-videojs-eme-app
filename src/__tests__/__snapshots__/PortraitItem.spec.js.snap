// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortraitItem renders correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  className="item"
  css={
    Object {
      "name": "qbkfep-PortraitItemStyles",
      "styles": "&.item{width:100%;border:5px solid;border-color:transparent;&:hover{border-color:yellow;}a{display:flex;}img{min-width:100%;max-width:100%;}};label:PortraitItemStyles;",
    }
  }
>
  <Link
    to={
      Object {
        "pathname": "/details/mulan-718ddc0b-8c5f-4afd-8071-13592e4e21a1",
        "state": Object {
          "from": "/",
        },
      }
    }
  >
    <Image
      src="https://images-0.wuaki.tv/system/artworks/134332/master/mulan-2020-1601313907.jpeg"
    />
  </Link>
</EmotionCssPropInternal>
`;
